torch/fx/_symbolic_trace.py

```diff
@@ -309,6 +309,20 @@ def __init__(
         # Mapping of node name to module scope
         self.node_name_to_scope: Dict[str, Tuple[str, type]] = {}
 
+    @compatibility(is_backward_compatible=True)
+    def get_fresh_qualname(self, prefix: str) -> str:
```

Comment:
what if you store a dict with the seen prefixes and then just bump the count if the prefix is in the dict? 

maybe premature optimization but could this while loop get pretty long in practice?

Reply:
yeah maybe. But all the existing examples just did this.

Reply:
We actually fixed a material compile time perf problem from exactly this pattern a while back, I can't find the diff, but please just fix the O(n^2) to not be n^2. A two line improvement is to maintain a counter and initialize i off of that.

Reply:
ok fine I updated it :)

Reply:
Unfortunately, this is causing a lot of annoyances for the thing I was worried about that led to me not adding it initially - it seems that in some cases the tracer state is persisted across different modules, causing a bunch of churn in various tests 🤔